### PR TITLE
Initial decomposition pass on matter.js

### LINKED ISSUE #156: Trivial: ByteArray thoughts
Some quick ideas r.e. ByteArray.

- Right now we monkeypatch Uint8Array, might be frowned upon by some library users
- Could we make ByteArray a subclass of Uint8Array instead of an alias, and put utility functions there?
- We could also add custom toString and toJSON methods so diagnostics like in #138 would show as hex rather than comma-separated decimal
- I could also add the base64 utilities using the codec I made for the security stuff

### PR DESCRIPTION
**Pull request by lauckhart**
This is a partial pass deconstructing the matter.js package.  I tried to stick to decisions we'd previously agreed to.

The new @project-chip/matter.js-general package contains those portions of matter.js that are not Matter specific.

The new @project-chip/matter.js-model package contains the Matter object model.

Since I was moving stuff around I got rid of the Uint8Array monkey patches we did in ByteArray.  For the most part this just consisted of replacing the "ByteArray" type with "Uint8Array".  The methods we injected I moved to a "Bytes" namespace containing utility functions.  This is a breaking change.

Following new conventions, the new packages don't have "matter.js" in their names, so they're just in "packages/general" and "packages/model".  I also renamed packages/matter.js-tools to packages/tools.  I did not rename the other directories yet.

I updated codegen and the intermediate model package dependencies, so codegen errors do not prevent codegen from running except for the generated model files in @project-chip/matter.js-model.

The new packages are smaller so I moved to a flat namespace for exports.  So there are no subpaths exported from the "general" package, even though most of the exports were previously exported as subpaths of matter.js.  I think this improves readability and makes consumption simpler.

"model" also has a flat namespace but this was true of the previous "@project-chip/matter.js/model" module so this isn't as big of a change.

I did my best to add exports to matter.js to support backward compatibility.  However I used a little editorial control to select which exports to reproduce at the old location...  I made sure to cover all of those we used previously in the examples.  Many of the other exports were probably not ever used externally.  Shouldn't be a big deal if I got it wrong; we can add additional exports or just inform users of the new locations.

Fixes #156

### CODE REVIEW

**Comment by lauckhart** [comment]
Rebased onto #1172

**Review by Apollon77 — COMMENTED** [review]
beside the small stuff found I just do not like the "multiple full -general package exports in node package" ... lets simply kill the re-exports on matter-node not and just export the new functionality. These reexports are "deprecated" anyway since some versions and with adding peer-deps in second step we move further .. so time now.

The more "on purpose exports" in matter.js are same-ish ... I have no idea how we want to manage that on this detail level in the future ... that wold be the ly idea to sub-structure "general" exports too "by dir" to have that easier ...

**Comment by Apollon77** [comment]
PS: And add infos to changelog please :-)

File: packages/general/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Matter.js support logic
+
+This package provides generic functionality required Matter.js that is not specific to Matter.
```
[the remark this answers is not in the document]
Comment (lauckhart) [reply packages/general/README.md]: done

File: packages/matter-node.js/src/exports/model.ts [hunk 2]
```diff
@@ -4,4 +4,4 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-export * from "@project-chip/matter.js/model";
+export * from "@project-chip/matter.js-model";
```
[the remark this answers is not in the document]
Comment (lauckhart) [reply packages/matter-node.js/src/exports/model.ts:7]: done

**Comment by lauckhart** [comment]
> PS: And add infos to changelog please :-)

Those are in the original commit, LMK if you think it needs more

**Comment by lauckhart** [comment]
> beside the small stuff found I just do not like the "multiple full -general package exports in node package" ... lets simply kill the re-exports on matter-node not and just export the new functionality. These reexports are "deprecated" anyway since some versions and with adding peer-deps in second step we move further .. so time now.

I'll do this in a separate commit refactoring matter-node.js

**Comment by lauckhart** [comment]
> The more "on purpose exports" in matter.js are same-ish ... I have no idea how we want to manage that on this detail level in the future ... that wold be the ly idea to sub-structure "general" exports too "by dir" to have that easier ...

These are a one-time backward-compat thing.  There are also changes in both directory layout and exported identifier names so they need a little finagling beyond straight reexports (e.g. a few files from "common" are now in general/src/util but most aren't).  Exporting the directories would make it slightly easier but I wanted to avoid because there wouldn't be any way to prevent people from using them.

**Review by Apollon77 — APPROVED** [review]

**Pull request merged**